google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -191,9 +191,20 @@ def ack_deadline(self):
         Returns:
             int: The ack deadline.
         """
-        target = min([self._last_histogram_size * 2, self._last_histogram_size + 100])
-        if len(self.ack_histogram) > target:
-            self._ack_deadline = self.ack_histogram.percentile(percent=99)
+        target_size = min(
+            self._last_histogram_size * 2, self._last_histogram_size + 100
+        )
+        hist_size = len(self.ack_histogram)
+
+        if hist_size > target_size:
+            self._last_histogram_size = hist_size
```

Comment:
I see an error here: `self._last_histogram_size` is used, but its value is never changed. According to the method docs, it should be done this way

Reply:
It looks like `self._last_histogram_size` value will be changed once the Manager has a non-trivial `self.ack_histogram`.